app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -295,6 +333,14 @@ def ca_cert
 
     private
 
+    def os_name_mapping(host)
+      host.operatingsystem.name =~ /redhat|centos/i ? 'rhel': host.operatingsystem.name
+    end
+
+    def arch_name_mapping(host)
+      host.architecture.name == 'x86_64' ? 'x64' : host.architecture.name if host.architecture
```

Comment:
@iNecas @ares I think architecture has to be present *always* in order to create a host connected to a ComputeResource - https://github.com/theforeman/foreman/blob/develop/app/models/host/managed.rb#L191 - how do you get to this point in the code w/o architecture being set?